CMSIS/DSP/Include/arm_math.h

```diff
@@ -1676,6 +1676,42 @@ __STATIC_INLINE q31_t arm_div_q63_to_q31(q63_t num, q31_t den)
                        ((((q31_t)x <<  8) >>  8) & (q31_t)0xFFFF0000)  ));
   }
 
+   /*
+   * @brief C custom defined SXTB16 with immediate rotate right 8 bits
+   */
+  __STATIC_FORCEINLINE uint32_t __SXTB16_ROR8(
```

Comment:
Where shall those SXTB16_RORn function be located? In CMSIS compiler headers or in math header?

Reply:
I would say CMSIS compiler header.

ACLE doesn't have it, it just has `__sxtb16` and `__ror`, but that's documenting compiler extensions, and as a compiler extension it's reasonable to expect the compiler to be able to combine the two. And it seems that ARMC5 and ARMC6 at least do.

But if forced to use assembler on at least one compiler, so no chance of optimisation, then an having an explicit `__SXTB16_ROR` makes sense alongside `__SXTB16`.

Reply:
I would add such functions to CMSIS compiler header.

Reply:
Here those functions are pure C fallback when no intrinsics are available.
In the compiler headers you have the corresponding versions with builtin or asm.

I don't think the pure C versions should move to the compiler headers.

Also, don't forget that now we also want to be able to compile CMSIS-DSP without using ANY of the compiler headers (for connecting to system like matlab or scipy which is really important for people designing signal processing code).

Reply:
Ah, these fallback ones shouldn't really be here at all. They should just be defined as __SXTB16(__ROR(op1, rotate)), just as for ARMC5 and ARMC6 (and IAR? which is missing from this PR?).

Everyone should be using that definition, really - only GCC (and maybe IAR) gains an optimisation by the 1-instruction assembly.

Reply:
I don't see this one __SXTB16_ROR8 on my version of arm_math.h on develop branch.
For this one, I agree it can be built from other ones.

For other, I disagree. We have the strong requirement that we need to be able to compile CMSIS-DSP without any reference to any of the existing compiler headers.

The version of the functions as close to a pure C version as possible are the m0 version (ARM_MATH_DSP off). But even in this case, there are a few remaining intrinsics in use in the code and so a C implementation of them is required.

This C implementation could be in a different header but this header would have to be in DSP/Includes. 

But this discussion is not really relevant for this pull request.

Reply:
I am specifically talking about having custom definitions for `_SXTB16_ROR` that manually unroll it, rather than using `__SXTB16(__ROR())`. I'm not talking about the other fallbacks.

Reply:
Ok. I agree.